packages/use-contractkit/src/ethers.ts

```diff
@@ -24,17 +25,22 @@ export const useProviderOrSigner = (): Web3Provider | JsonRpcSigner => {
 };
 
 export const useGetConnectedSigner = (): (() => Promise<JsonRpcSigner>) => {
-  const { kit, getConnectedKit } = useContractKit();
+  const { kit, getConnectedKit, network } = useContractKit();
   const signer = useProviderOrSigner();
+  const { chainId, name } = network;
+
   return useCallback(async () => {
     if (kit.defaultAccount) {
       return signer as JsonRpcSigner;
     }
+
     const nextKit = await getConnectedKit();
     const nextProvider = nextKit.web3
       .currentProvider as unknown as ExternalProvider;
-    return new Web3Provider(nextProvider).getSigner(nextKit.defaultAccount);
-  }, [kit.defaultAccount, getConnectedKit, signer]);
+    return new Web3Provider(nextProvider, { chainId, name }).getSigner(
```

Comment:
anyone know why this needs to be a new provider? why cant we reuse provider from `userProvider`? 

@dckesler @jmrossy

Reply:
@nambrot

Reply:
@AlexBHarley would probably be the best person to ask. I haven't reviewed this part thoroughly but it's entirely possible it was an oversight. No disrespect to Alex who overall did a great job but some parts were written hastily I think!

Reply:
Looking at it a bit more, do you mean reuse the kit's provider? That's a Web3 provider, this creates an ethers.js provider

Reply:
`new Web3Provider(nextProvider, { chainId, name })` looks to give us same kind of thing as `useProvider` no?

Reply:
I'm not quite understanding you I think. This is **inside** the useProvider hook.

Reply:
This PR is modifying 2 places. the `useProvider` and inside `useGetConnectedSigner` this line is inside the later